### DISCUSSION TITLE
I2S function Japanese format

### DISCUSSION THREAD

**Discussion by antalpeter** [micropython/micropython#11447] [RP2040 / Pico]
Is the i2S function support japanese format? Most of the CD use that.

**Comment by miketeachman** [comment]
The I2S protocol is agnostic to any language.  I2S is based on digital audio samples as described in the [MicroPython I2S docs](https://docs.micropython.org/en/latest/library/machine.I2S.html).  For examples on how to use the I2S protocol with WAV files you might want to look at an [I2S examples repo](https://github.com/miketeachman/micropython-i2s-examples) that I maintain.  Maybe you can convert the CD audio content to WAV files?  I hope this helps.

**Reply by antalpeter** [reply]
Dir Mike!
What I need is a actual **format** and **mode** parameter available for the micropython I2S function for CD audio or Japanese format

**Reply by miketeachman** [reply]
I'm a bit confused on what you are asking for.  Is it possible to attach an example of a file in the format you are working with?

**Reply by scruss** [reply]
is it [Least significant bit justified](https://github.com/earlephilhower/arduino-pico/issues/1094) (LSBJ) format, perhaps?
